extensions/amp-consent/0.1/consent-state-manager.js

```diff
@@ -277,6 +294,29 @@ export class ConsentInstance {
     });
   }
 
+  /**
+   * Set the values of the purposes.
+   * @param {!Object<string, boolean>} purposeMap
+   * @param {boolean=} opt_defaultsOnly
+   */
+  updatePurposes(purposeMap, opt_defaultsOnly) {
+    if (!this.purposeConsents_) {
+      this.purposeConsents_ = {};
+    }
+    const purposes = Object.keys(purposeMap);
+    for (let i = 0; i < purposes.length; i++) {
```

Comment:
nit: look into using `for..of` for such cases - it improves readability and you don't have to do `purpose = purpose[i]` e.g.
```
for (const purpose of purposes) {
  if (opt_defaultsOnly && this.purposeConsents_[purpose]) {
        continue;
  }
}
```

Reply:
`for..of` aren't allowed due to the large polyfill it requires. I can change it to `forEach` though for readability.

Reply:
Ah never mind. Leave it as is.